### PR TITLE
refactor: rename `@bjoerge/mutiny` to `@sanity/mutate` and setup automation

### PR DESCRIPTION
**Pull request by stipsan**
This PR renames all references to `@bjoerge/mutiny` to `@sanity/mutate`.
The README.md now includes a clear warning that this is an experimental package, Release Please automation is also setup to stay on `v0` to signal this as well.
The Release Please automation has been tested, and used to publish [v0.7.1](https://www.npmjs.com/package/@sanity/mutate/v/0.7.1) already.
Moving forward breaking changes will bump minors, like in this test PR: https://github.com/sanity-io/mutate/pull/10

Once we feel this is ready for userland we can start bumping regular majors, and make the first v1 release, by removing `bump-minor-pre-major` from `release-please-config.json`.

GitHub Actions have been refactored to follow our latest best practices for security, by declaring `permissions` up front, and perf, by declaring `concurrency` to auto cancel concurrent PR builds and by using `on.push` and `on.pull_request` configs that don't lead to actions running twice per commit on a PR, and a new secret `NPM_SANITYPE_TOKEN` that only have access to installing `@sanity/sanitype` and no other private packages in our org.

Other changes:
- moved the prettier config to use `@sanity/prettier-config`, with a few select overrides.
- added `CHANGELOG.md` generated from the previous commit history, it may have entries that no longer makes sense, but that can be fixed after this is merged and doesn't need to be a blocker.
- added `LICENSE` that follows the same format as the rest of our official packages.
- bumped TypeScript to 5.5, pinning it since TS does not follow semver. 5.6 may very well pull in breaking changes and it's important that we can at any time nuke `node_modules` and `pnpm-lock.yaml` and reinstall without accidentally pulling in a new minor of TSC that contains breaking changes.

### CODE REVIEW

File: .npmrc [hunk 1]
```diff
@@ -0,0 +1,2 @@
+; This is needed for prettier to be able to use the plugin specified by the `@sanity/prettier-config` preset
+public-hoist-pattern[]=prettier-plugin-packagejson
```
Comment (stipsan) [inline .npmrc:2]: I'll see if I can resolve this within `@sanity/prettier-config` itself, but for now this is necessary 🤷

File: .prettierignore [hunk 2]
```diff
@@ -1 +1,4 @@
 /dist
+CHANGELOG.md
```
Comment (stipsan) [inline .prettierignore:2]: Release Please doesn't follow prettier formatting rules, so it becomes annoying to have Prettier constantly formatting a auto generated file

File: .prettierrc.cjs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+const preset = require('@sanity/prettier-config')
+
+module.exports = {
+  ...preset,
+  printWidth: 80,
```
Comment (stipsan) [inline .prettierrc.cjs:5]: The `printWidth` in `@sanity/prettier-config` is `100`

File: .prettierrc.cjs [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+  arrowParens: 'avoid',
```
Comment (stipsan) [inline .prettierrc.cjs:6]: `@sanity/prettier-config` doesn't set this option.

File: .prettierrc
Comment (stipsan) [inline .prettierrc:1]: `@sanity/prettier-config` have almost all these options already

**Review by bjoerge — APPROVED** [review]
Fantastic! 💖

**Pull request merged**